SocketIOClientSwift/SocketIOClient.swift

```diff
@@ -68,45 +68,41 @@ public final class SocketIOClient: NSObject, SocketEngineClient {
         self.socketURL = socketURL
         self.opts = opts
         
-        if let connectParams = opts?["connectParams"] as? [String: AnyObject] {
-            self.connectParams = connectParams
-        }
-        
-        if let logger = opts?["logger"] as? SocketLogger {
-            Logger = logger
-        }
-        
-        if let log = opts?["log"] as? Bool {
-            Logger.log = log
-        }
-        
-        if let nsp = opts?["nsp"] as? String {
-            self.nsp = nsp
-        }
-        
-        if let reconnects = opts?["reconnects"] as? Bool {
-            self.reconnects = reconnects
-        }
-        
-        if let reconnectAttempts = opts?["reconnectAttempts"] as? Int {
-            self.reconnectAttempts = reconnectAttempts
-        } else {
-            self.reconnectAttempts = -1
-        }
-        
-        if let reconnectWait = opts?["reconnectWait"] as? Int {
-            self.reconnectWait = abs(reconnectWait)
-        }
-        
-        if let handleQueue = opts?["handleQueue"] as? dispatch_queue_t {
-            self.handleQueue = handleQueue
-        } else {
-            self.handleQueue = dispatch_get_main_queue()
+        for option in opts ?? [] {
+            switch option {
+            case .ConnectParams(let params):
+                connectParams = params
+            case .Reconnects(let reconnects):
+                self.reconnects = reconnects
+            case .ReconnectAttempts(let attempts):
+                reconnectAttempts = attempts
+            case .ReconnectWait(let wait):
+                reconnectWait = wait
+            case .Nsp(let nsp):
+                self.nsp = nsp
+            case .Log(let log):
+                Logger.log = log
+            case .Logger(let logger):
+                Logger = logger
+            case .HandleQueue(let queue):
+                handleQueue = queue
+            default:
```

Comment:
It's needed because in Swift switches must be exhaustive. And because we don't care about every option possible here, we need a default.